### DISCUSSION TITLE
New Mastodon API client library for PHP

### DISCUSSION THREAD

**Discussion by vazaha-nl** [mastodon/mastodon#26451] [Show and tell]
Hi community,

I have been working very hard on my brand new [Mastodon API client for PHP](https://github.com/vazaha-nl/mastodon-api-client) and today I am very happy to announce the release of version 1.0. 

There are already a few other API clients for PHP available but they are either not complete, not actively maintained or lacking in other aspects. So decided to build my own, and do a proper job. I wanted this library to be fully typed, and complete: every api method and entity should be implemented. This turned out to be quite a challenge.

It was too time intensive to implement everything by hand. And there is no complete / up to date openapi spec available. So I had to find another way. I tried to [generate a working spec using AppMap](https://appmap.io/blog/2022/12/14/automatically-generating-openapi-docs-for-mastodon/), but unfortunately that was unsuccessful due to [a bug](https://github.com/getappmap/appmap-ruby/issues/340). In the end I decided to create a parser for the [Mastodon markdown documentation](https://github.com/mastodon/documentation). Which was not very easy but it turned out to be possible and feasible.

Check out the code, documentation and usage examples at https://github.com/vazaha-nl/mastodon-api-client. It would be great if my library could be included in the list at https://docs.joinmastodon.org/client/libraries/

Looking forward to your feedback!

Best regards,
Lennart

**Comment by andypiper** [comment]
This is great @vazaha-nl - thank you for sharing! We are aware of the pain that we don't currently have an OpenAPI specification of our own, it may be worth knowing that if you are only working from the documentation, you may currently be [missing some of the available API endpoints and features](https://github.com/mastodon/documentation/issues/1211), but I'm working in the background to improve things here.

I'll definitely look at getting your library added to the docs site.
(update - now [included](https://docs.joinmastodon.org/client/libraries/#php)!)

**Comment by vazaha-nl** [comment]
Thanks a lot Andy!

Once the currently undocumented endpoints enter the documentation, it should be easy to update my client code. Will keep an eye on it!
